gold-phone-input.html

```diff
@@ -181,51 +194,122 @@
       // If there's an initial input, validate it.
       if (this.value)
         this._handleAutoValidate();
+      
+    },
+
+    attached: function() {
+
+      if(this.noCountryCode) {
+        this.set('countryCode', '');
+        Polymer.dom(this).node.querySelector('label').classList.add('label-no-country-code');
+      }
+
+      this.set('maxlength', this.phoneNumberPattern.length);
+
     },
 
     _phoneNumberPatternChanged: function() {
       // Transform the pattern into a regex the iron-input understands.
       var regex = '';
       regex = this.phoneNumberPattern.replace(/\s/g, '\\s');
+      regex = regex.replace(/\(/gi, '\\(');
+      regex = regex.replace(/\)/gi, '\\)');
       regex = regex.replace(/X/gi, '\\d');
       regex = regex.replace(/\+/g, '\\+');
       this.$.input.pattern = regex;
+
+      this.regex = regex;
     },
 
     /**
      * A handler that is called on input
      */
     _onValueChanged: function(value, oldValue) {
+
       // The initial property assignment is handled by `ready`.
-      if (oldValue == undefined || value === oldValue)
+      if (oldValue == undefined || value == undefined) {
+        this.value = '';
```

Comment:
The "setting the value to undefined resets it to the empty string" test fails without this.